### ISSUE TITLE
Keep scroll position on ViewScale Change, Auto Expand or Add Task

### ISSUE THREAD

**Issue by Schweigi** [angular-gantt/angular-gantt#181]
The current horizontal scroll position gets changed when on of the following things happens:
- ViewScale change (e.g. switch from `day` to `hour`)
- Auto Expand (e.g. because the user scrolled to the left gantt side)
- Add Task (e.g. because a new task is added and the gantt date range is expanded)

The component `keepScrollPos` (https://github.com/angular-gantt/angular-gantt/blob/master/src/ui/util/keepScrollPos.factory.js) can be used to wrap code which potentially changes the scroll position.  After any change the component will then restore the original position. Unfortunately this component doesn't work anymore because the code how and when columns are generated has changed.

After a little investigation I think this component is not needed anymore because it should be possible to add some improved code to keep the scroll position at one single point when the columns are generated in the function `generateColumns` (https://github.com/angular-gantt/angular-gantt/blob/master/src/logic/gantt.factory.js#L151). This should then cover all the cases. 

So in conclusion this would mean every time the function `generateColumns` is called the scroll position is restored to what it was before. E.g. when the gantt is at scroll position `16th Sept` and is switched to `hour` scale the scroll position is still `16th Sept`.

I would suggest to not use anymore the same approach as the current keepScrollPos component by trying to memorize the scrollLeft/scrollWidth properties. Instead before `generateColumns` does any changes the date&time at the screen center is stored and then restored after all the changes have been made.

Any other suggestions on how to solve this? Otherwise I will do an implementation for this.
